myhoard/backup_stream.py

```diff
@@ -455,12 +456,17 @@ def run(self):
 
                 self.wakeup_event.wait(self._get_iteration_sleep())
                 self.wakeup_event.clear()
+                consecutive_errors = 0
             except Exception as ex:  # pylint: disable=broad-except
                 self.log.exception("Unexpected exception in mode %s", self.mode)
                 self.stats.unexpected_exception(ex=ex, where="BackupStream.run")
                 self.state_manager.increment_counter(name="backup_errors")
                 self.stats.increase("myhoard.backup_stream.errors")
-                time.sleep(1)
+                sleep_time = min(1.5 ** consecutive_errors, 30.0)
+                # Don't increase the counter indefinitely or else we'll get to exponent that cannot be handled anymore
+                if consecutive_errors < 20:
+                    consecutive_errors += 1
```

Comment:
nit. I'd rather avoid a "lying variable" that does not really count the errors and move the clamp to the computation instead :
```suggestion
                # Clamp the exponent to avoid OverflowError
                sleep_time = min(1.5 ** min(20, consecutive_errors), 30.0)
                consecutive_errors += 1
```

Reply:
Changed accordingly